### ISSUE TITLE
Fails on prepared queries.

### ISSUE THREAD

**Issue by thepug** [kennethklee/node-cassandra-driver-mock#1]
I get the following error when trying to run client.execute with the prepare: true option.

```
Uncaught TypeError: Cannot call method 'getPreparedInfo' of undefined
    at Client._getPrepared (/Users/nathan/swdevl/path/messaging-proto-services/session/node_modules/cassandra-driver/lib/client.js:563:28)
    at /Users/nathan/swdevl/path/messaging-proto-services/session/node_modules/cassandra-driver/lib/client.js:393:12
    at fn (/Users/nathan/swdevl/path/messaging-proto-services/session/node_modules/cassandra-driver/node_modules/async/lib/async.js:641:34)
    at Object._onImmediate (/Users/nathan/swdevl/path/messaging-proto-services/session/node_modules/cassandra-driver/node_modules/async/lib/async.js:557:34)
    at processImmediate [as _immediateCallback] (timers.js:345:15)
```

function call in coffeescript `client.execute UPSERT_REGION, params, {prepare: true}, (err, result) -> reply null`

changing prepare to false works great.

**Comment by thepug** [comment]
strange.  I see that you have tests for prepare: true and they run correctly.  I'll try to gather more details on why my setup fails.

**Comment by kennethklee** [comment]
It appears you're using cassandra-driver v2.0.0 which is fairly new; it's about a month old.

There's been a whole bunch of changes to how queries are prepared before being executed. I tried to run the tests on v2 and found that we'd need to mock a connection and reference the metadata from the control-connection to the client.

I'll see what I can do. Feel free to fork and send a PR my way for v2 of the driver if you need this urgently.

**Comment by thepug** [comment]
https://github.com/kennethklee/node-cassandra-driver-mock/pull/2

**Comment by kennethklee** [comment]
@thepug, see PR #3

**Comment by kennethklee** [comment]
Published on npm, v2.0.0
